File: composer/src/main/java/io/piano/android/composer/model/DelayBy.java
package io.piano.android.composer.model;

import androidx.annotation.Keep;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

@Keep
public class DelayBy {
    @Nullable
    @SerializedName("type")
    public DelayType type;

    @SerializedName("value")
    public int value;

    public static enum DelayType {
        @SerializedName("time")
        TIME,
        @SerializedName("scroll")
        SCROLL
    }

    public boolean isDelayedByTime() {
        return type == DelayType.TIME && value > 0;
    }

    public boolean isDelayedByScroll() {
        return type == DelayType.SCROLL && value > 0;
    }
}

File: composer/src/main/java/io/piano/android/composer/model/DelayBy.kt
package io.piano.android.composer.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
public class DelayBy(
    @JvmField public val type: DelayType?,
    @JvmField public val value: Int,
) {
    @JsonClass(generateAdapter = false)
    public enum class DelayType {
        @Json(name = "time")
        TIME,

        @Json(name = "scroll")
        SCROLL,
    }

    public val isDelayedByTime: Boolean
        get() = type == DelayType.TIME && value > 0

    public val isDelayedByScroll: Boolean
        get() = type == DelayType.SCROLL && value > 0
}
